torch/_refs/nn/functional/__init__.py

```diff
@@ -262,6 +263,36 @@ def relu(a: TensorLikeType, inplace: bool = False) -> TensorLikeType:
     return torch.where(torch.le(a, 0), 0, a)
 
 
+@register_decomposition(aten.channel_shuffle)
+@out_wrapper()
+def channel_shuffle(input: TensorLikeType, groups: int) -> TensorLikeType:
+    """
+    Reference implementation of :func:`torch.nn.functional.channel_shuffle`.
+    """
+    from torch._meta_registrations import device_hint
+
+    torch._check(
+        groups > 0,
+        lambda: f"Number of groups to divide channels in must be positive. Value of groups:{groups}",
+    )
```

Comment:
Can you add the other two checks in as well:
https://github.com/pytorch/pytorch/blob/367213a608528ee74e67e03bf11f775e263ef480/aten/src/ATen/native/ChanelShuffle.cpp#L39-L48